api/v1beta2/kustomization_types.go

```diff
@@ -0,0 +1,361 @@
+/*
+Copyright 2021 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	"time"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/fluxcd/pkg/apis/kustomize"
+	"github.com/fluxcd/pkg/apis/meta"
+	"github.com/fluxcd/pkg/runtime/dependency"
+)
+
+const (
+	KustomizationController   = "kustomize-controller"
```

Comment:
What is the reason this is included in the API package? Has it any special meaning to a consumer? Otherwise I think it is a controller implementation detail and does not belong in the API.

Reply:
It's used for the managed fields owner name. I think this must be part of the API as if this value changes, the controller will conflict with itself.

Reply:
Yes, but the managed fields owner name is set by the controller. I would thus expect this value to be part of the application (contract), and not of the API that can be consumed (and theoretically, also driven) by others.

Reply:
Isn't the finalizer part of the application contract? I'm for moving both out the API.

Reply:
Will need to confirm this based on some sources that I don't have at hand right now, will get back to you about this on Monday (with references).

Reply:
Finalizers do seem to be part of the API: https://pkg.go.dev/sigs.k8s.io/cluster-api@v0.3.10-rc.1/exp/addons/api/v1alpha3#pkg-constants